src/Chainweb/Pact/TransactionExec.hs

```diff
@@ -261,13 +258,26 @@ applyCoinbase v logger dbEnv (Miner mid mks) reward@(ParsedDecimal d) pd parentH
     forkTime = vuln797FixDate v
     fork1_3InEffect = blockTime >= forkTime
     throwCritical = fork1_3InEffect || enfCBFailure
+
     blockTime = blockTimeOf pd
+        -- FIXME it should hold that @blockTime == _blockCreationTime parentHeader@, but
+        -- in some unit test runs that is not the case
+
+    bh = fromIntegral $ _pdBlockHeight pd
+
+    cid = V._chainId parentHeader
+        -- FIXME it should hold that
+        -- @cid == unsafeFromText $ P._chainId $ _pmChainId $ _pdPublicMeta pd@
+        -- but in some unit test runs that is not the case. The chain id in public data is empty.
```

Comment:
Up until now, we have being using a default value for public metadata for coinbase transactions. We can fix this in the upcoming hardfork, but not much is gained, since it is a miner-only transaction, and will never make use of any metadata values.

Reply:
If we stop using PublicData this problem goes away correct?

Reply:
Yes